File: src/Listas2/pratica/Questao05.java
package Listas2.pratica;

import java.util.Scanner;

class Circulo {

    private int x, y, raio;
    public static final double PI = 3.14159;

    public Circulo(int x, int y, int raio) {
        this.x = x;
        this.y = y;
        this.raio = raio;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public int getRaio() {
        return raio;
    }

    public void setRaio(int raio) {
        this.raio = raio;
    }

    public double circunferencia() {
        return 2 * PI * raio;
    }
}

public class Questao05 {  //use modificador de acesso (public) apenas para a classe principal

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        Circulo[] vetor1 = new Circulo[3];
        for (int i = 0; i < 3; i++) {
            int x = sc.nextInt();
            int y = sc.nextInt();
            int raio = sc.nextInt();
            vetor1[i] = new Circulo(x, y, raio);
        }

        Circulo[] vetor2 = vetor1;

        Circulo[] vetor3 = new Circulo[3];
        for (int i = 0; i < 3; i++) {
            vetor3[i] = new Circulo(
                    vetor1[i].getX(),
                    vetor1[i].getY(),
                    vetor1[i].getRaio()
            );
        }

        for (int i = 0; i < 3; i++) {
            int raio = sc.nextInt();
            vetor1[i].setRaio(raio);
        }

        System.out.println("vetor1:");
        for (Circulo circulo : vetor1) {
            System.out.printf("%d %d %d\n", circulo.getX(), circulo.getY(), circulo.getRaio());
        }

        System.out.println("vetor2:");
        for (Circulo circulo : vetor2) {
            System.out.printf("%d %d %d\n", circulo.getX(), circulo.getY(), circulo.getRaio());
        }

        System.out.println("vetor3:");
        for (Circulo circulo : vetor3) {
            System.out.printf("%d %d %d\n", circulo.getX(), circulo.getY(), circulo.getRaio());
        }
    }
}

File: Listas C++/Lista 2/questao05.cpp
#include <iostream>
#include <vector>
#include <cmath> 
#include <functional>

class Circulo {
private:
    int x;
    int y;
    int raio;

public:
    static constexpr double PI = 3.14159;

    Circulo(int x, int y, int raio) : x(x), y(y), raio(raio) {}

    int getX() const {
        return x;
    }

    int getY() const {
        return y;
    }

    int getRaio() const {
        return raio;
    }

    void setRaio(int raio) {
        this->raio = raio;
    }

    double circunferencia() const {
        return 2 * PI * raio;
    }
};

int main() {
    
    std::vector<Circulo> vetor1;
    vetor1.reserve(3);

    for (int i = 0; i < 3; i++) {
        int x, y, raio;
        if (!(std::cin >> x >> y >> raio)) return 1;
        vetor1.emplace_back(x, y, raio);
    }

    std::vector<std::reference_wrapper<Circulo>> vetor2;
    for (Circulo& c : vetor1) {
        vetor2.push_back(std::ref(c));
    }
    
    std::vector<Circulo> vetor3;
    vetor3.reserve(3);
    
    for (int i = 0; i < 3; i++) {
        vetor3.emplace_back(
            vetor1[i].getX(),
            vetor1[i].getY(),
            vetor1[i].getRaio()
        );
    }

    for (int i = 0; i < 3; i++) {
        int novoRaio;
        if (!(std::cin >> novoRaio)) return 1;
        vetor1[i].setRaio(novoRaio);
    }

    std::cout << "vetor1:\n";
    for (const auto& circulo : vetor1) {
        std::cout << circulo.getX() << " " << circulo.getY() << " " << circulo.getRaio() << "\n";
    }

    std::cout << "vetor2:\n";
    for (const auto& circuloRef : vetor2) { 
        std::cout << circuloRef.get().getX() << " " << circuloRef.get().getY() << " " << circuloRef.get().getRaio() << "\n";
    }

    std::cout << "vetor3:\n";
    for (const auto& circulo : vetor3) {
        std::cout << circulo.getX() << " " << circulo.getY() << " " << circulo.getRaio() << "\n";
    }

    return 0;
}